Migrate from Dockerhub to Amazon ECR.

MarkAZhang opened the pull request:
# Description

* Change the travis-ci build/push scripts and deployment scripts to use ECR for the docker registry.

# Notes
* Created new idseq-web ECR repo.
* Also needed to add appropriate permissions to travis-ci IAM role. These permissions are scoped to only the newly created idseq-web ECR repo.
* AWS authentication on Travis CI relies on Travis CI repository variables already set on the idseq-web repo in Travis CI. The only variable we needed to add was the aws account id.

# Tests

* Verified that Travis CI builds successfully. Verified that Travis CI properly uses cached image on subsequent builds (and speeds up from 8m to 3m).
* Verified that deployment to sandbox environment succeeds.

jshoe commented on .travis.yml:
```diff
@@ -82,9 +82,16 @@ jobs:
         - npm test
     - name: "Docker Build"
-      language: shell
+      language: python
```
was this supposed to change?

MarkAZhang replied:
I had changed it thinking we needed python for "pip install". It turns out that travis CI includes python with the "minimal/shell" build anyways. I will change it back and see what happens.

jshoe commented on bin/push-docker:
```diff
@@ -11,25 +11,6 @@ echo $REPO
-
 # Retry up to 5 times total if failed Docker push
 run=1
 count=1
```
While you're at it, could change these comments below to say something like "Trying ECR push"/"Exceeded retries with ECR"

MarkAZhang replied:
Good catch!

jshoe commented on bin/deploy:
```diff
@@ -63,6 +63,7 @@ end
 
 def post_slack_message(payload, slack_hook)
   puts `curl -X POST --data-urlencode payload=#{Shellwords.escape(JSON.dump(payload))} #{slack_hook}`
+
```
nit: extra blank line

MarkAZhang commented:
Also fixed an issue brought up by Tiago where I was storing `AWS_ACCOUNT_ID` in an environment variable in the Terminal and not removing it after the script was run.

Changed to using backticks, which allows you to get the output of a shell command in Ruby.
